static/app/components/events/interfaces/searchBarActionV2.tsx

```diff
@@ -0,0 +1,119 @@
+import styled from '@emotion/styled';
+
+import DropdownButtonV2 from 'sentry/components/dropdownButtonV2';
+import CompactSelect from 'sentry/components/forms/compactSelect';
+import SearchBar from 'sentry/components/searchBar';
+import {t, tn} from 'sentry/locale';
+import space from 'sentry/styles/space';
+
+type FilterOption = React.ComponentProps<typeof CompactSelect>['options'][0];
+
+type Props = {
+  onChange: (value: string) => void;
+  placeholder: string;
+  query: string;
+  className?: string;
+  filterOptions?: FilterOption[];
+  filterSelections?: FilterOption[];
+  onFilterChange?: (options: FilterOption[]) => void;
+};
+
+function SearchBarAction({
+  onChange,
+  query,
+  placeholder,
+  filterOptions,
+  filterSelections,
+  onFilterChange,
+  className,
+}: Props) {
+  function trigger({props, ref}) {
```

Comment:
can we type this better?

maybe

```tsx
const trigger: React.ComponentProps<typeof CompactSelect>['trigger'] = ({props, ref}) => (
  <StyledTrigger
    size="small"
    priority={filterSelections && filterSelections.length > 0 ? 'primary' : 'default'}
    ref={ref}
    {...props}
  >
    {filterSelections?.length
      ? tn('%s Active Filter', '%s Active Filters', filterSelections.length)
      : t('Filter By')}
  </StyledTrigger>
);

Reply:
Ahh when doing that, I discovered a typing error in `DropdownButtonV2` (fixed here: https://github.com/getsentry/sentry/pull/35263) 😛